### PR TITLE
Add Prettier

### PR DESCRIPTION
**Pull request by denniskigen**
This PR introduces [Prettier](https://prettier.io) for standardized code formatting. It adds a formatting step and a pre-commit hook and configures it so that the prettier formats staged files before pre-commit.

### CODE REVIEW

File: src/app/Services/openmrs-api/save-form.service.ts [hunk 1]
```diff
@@ -1,170 +1,188 @@
...
-    ///////////////////////////////////////////////////////////////////////
+  getNewDescription() {
+    return this.newDescription.asObservable();
+  }
 
-    publish(uuid) {
-        const body = { published : true };
-        return this.http.post(`${this.baseUrl}/ws/rest/v1/form/${uuid}`, body, {headers: this.headers});
-    }
+  ///////////////////////////////////////////////////////////////////////
```
Comment (corneliouzbett) [inline src/app/Services/openmrs-api/save-form.service.ts:141]: You should remove this. It's not neccessary

**Review by corneliouzbett — APPROVED** [review]
LGTM!

BTW - What does `replace.build.js` file do?

**Comment by denniskigen** [comment]
> LGTM!
> 
> BTW - What does `replace.build.js` file do?

It creates a timestamp build version string like this: `build v-1.0 20/03/2018 09:45:31` that is displayed in the bottom right hand side of the page footer.

File: src/app/Services/openmrs-api/save-form.service.ts (hunk 1 above)
Comment (denniskigen) [reply src/app/Services/openmrs-api/save-form.service.ts:141]: Done

**Pull request merged**